streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -53,34 +85,121 @@ public void process(final K key, final Change<V> value) {
                 internalProcessorContext.forward(key, value);
             } // else skip
         } else {
-            throw new NotImplementedException();
+            ensureBuffer();
+            buffer(key, value);
+            enforceSizeBound();
         }
     }
 
-    private boolean shouldForward(final Change<V> value) {
-        return !(value.newValue == null && suppress.suppressTombstones());
+    @SuppressWarnings("unchecked")
+    private <T> Serde<T> castSerde(final Serde<?> untyped, final String keyOrValue) {
+        try {
+            return (Serde<T>) untyped;
+        } catch (final ClassCastException e) {
+            throw new TopologyException(
+                "Unable to use the configured default " + keyOrValue +
+                    " serde. Ensure that the correct " + keyOrValue + " serde is set upstream of " + myNode + ".",
+                e
+            );
+        }
     }
 
     private long definedRecordTime(final K key) {
         return suppress.getTimeDefinition().time(internalProcessorContext, key);
     }
 
+    private void buffer(final K key, final Change<V> value) {
+        final long time = suppress.getTimeDefinition().time(internalProcessorContext, key);
```

Comment:
`suppress.getTimeDefinition()` should return the same thing each time? Should we put it into a member variable?